packages/keystatic/src/app/auth.ts

```diff
@@ -20,23 +21,32 @@ export function getSyncAuth(config: Config) {
     return { accessToken };
   }
   if (config.storage.kind === 'cloud') {
-    const unparsedTokenData = localStorage.getItem(
-      'keystatic-cloud-access-token'
-    );
-    let tokenData;
-    try {
-      tokenData = storedTokenSchema.parse(JSON.parse(unparsedTokenData!));
-    } catch (err) {
-      return null;
-    }
-    if (!tokenData || tokenData.validUntil < new Date()) {
-      return null;
-    }
-    return { accessToken: tokenData.token };
+    return getCloudAuth(config);
   }
   return null;
 }
 
+export function getCloudAuth(config: Config) {
+  if (!config.cloud?.project) return null;
+  const unparsedTokenData = localStorage.getItem(
+    'keystatic-cloud-access-token'
+  );
+  let tokenData;
+  try {
+    tokenData = storedTokenSchema.parse(JSON.parse(unparsedTokenData!));
+  } catch (err) {
+    return null;
+  }
+  if (
+    !tokenData ||
+    tokenData.validUntil < new Date() ||
+    tokenData.project !== config.cloud.project
```

Comment:
This is so that we don't try using a token that is for a different project than it was created for which is pretty likely to happen running different projects locally over time on the same port